src/structs.jl

```diff
@@ -14,12 +14,23 @@ function read(str::AbstractString, ::Type{T}; kw...) where {T}
     pos = 1
     b = getbyte(buf, pos)
     @wh
-    pos, x = read(StructType(T), buf, pos, len, b, T; kw...)
-    return x
+    return buf, pos, len, b
 @label invalid
     invalid(error, buf, pos, T)
 end
 
+function read(str::AbstractString, ::Type{T}; kw...) where {T}
+    buf, pos, len, b = _buf_pos_len_b(str, T)
+    pos, x = read(StructType(T), buf, pos, len, b, T; kw...)
+    return x
+end
+
+function read!(str::AbstractString, x::T; kw...) where {T}
+    buf, pos, len, b = _buf_pos_len_b(str, T)
```

Comment:
I think for now, we should add an assert that `StructType(T) === StructTypes.Mutable()` since this isn't going to work/fail in weird ways for other struct types. Can we do that and add a test or two showing the throw for other struct types?

Reply:
Ah, I see you did actually add some tests for immutable; still, I'd rather throw an `ArgumentError` than fallback to the `MethodError` being thrown in the tests.

Reply:
Done. It now throws an `ArgumentError` instead of a `MethodError`.